bin/build.ts

```diff
@@ -0,0 +1,48 @@
+import util from 'util'
+import { remove, ensureDir } from 'fs-extra'
+import ParcelBundler, { ParcelOptions } from 'parcel-bundler'
+import { generateSW } from 'workbox-build'
+const exec = util.promisify(require('child_process').exec)
+
+async function BuildUI() {
+  const options: ParcelOptions = {
+    outDir: 'dist/ui',
+    minify: true,
+    watch: false,
+    contentHash: true
+  }
+  const JSOptions: ParcelOptions = {
+    outDir: 'dist/public',
+    minify: true,
+    watch: false,
+    contentHash: false
+  }
+
+  await remove('dist/')
+  await ensureDir('dist/public')
+
+  await exec('cp -r public/* dist/public/ && cp package* dist && tsc -b functions/tsconfig.json')
+  await generateSW({
+    swDest: 'dist/public/service-worker.js',
+    importScripts: ['worker.js'],
+    globDirectory: '.',
+    runtimeCaching: [
+      {
+        urlPattern: /^https?.*/,
+        handler: 'NetworkFirst',
+        options: {
+          cacheName: 'offlineCache',
+          expiration: {
+            maxEntries: 200,
+          },
+        },
+      },
+    ],
+  })
+  const JSBundler = new ParcelBundler(['public/assets.urls'], JSOptions)
+  const UIBundler = new ParcelBundler(['ui/server.tsx'], options)
```

Comment:
These should prob be named server and client, rather than js and ui, which doesn't quite make sense to me.